viper/src/ast_factory/expression.rs

```diff
@@ -198,29 +197,85 @@ impl<'a> AstFactory<'a> {
         self.int_lit_from_ref_with_pos(i, self.no_position())
     }
 
-    pub fn backend_bv_lit(&self, bits: u64) -> Expr<'a> {    
+    // Backend Bitvectors
+    pub fn backend_bv32_lit(&self, bits: u32) -> Expr<'a> {    
+        let bv_factory_ = ast::utility::BVFactory::with(self.env);
+        let bv_factory = ast::utility::BVFactory::new(&bv_factory_, 32).unwrap();
+        let from_int = ast::utility::BVFactory::call_from__int(&bv_factory_, bv_factory, self.jni.new_string("toBV32")).unwrap();
+        self.backend_func_app(from_int, &[self.int_lit(bits as i64)], self.no_position())
+    }
+
+    pub fn backend_bv64_lit(&self, bits: u64) -> Expr<'a> {    
         let bv_factory_ = ast::utility::BVFactory::with(self.env);
         let bv_factory = ast::utility::BVFactory::new(&bv_factory_, 64).unwrap();
         let from_int = ast::utility::BVFactory::call_from__int(&bv_factory_, bv_factory, self.jni.new_string("toBV64")).unwrap();
         self.backend_func_app(from_int, &[self.int_lit(bits as i64)], self.no_position())
     }
+    
+    // Backend Floating-Points
+    pub fn float_binop(&self, op_code: u8, f_size: u8, left: Expr, right: Expr) -> Expr<'a> {
```

Comment:
Could you please add `FloatOpKind` enum to `viper` crate and use it instead of `u8`?